src/index.ts

```diff
@@ -248,9 +259,49 @@ export interface CanonicalPurpose {
   purposeCodes?: string[]
 }
 
+// IdentityLocation is the location on the page from which to retrieve identity information
+export enum IdentityLocation {
+  IDENTITY_LOCATION_UNDEFINED,
+  IDENTITY_LOCATION_DATA_LAYER,
+  IDENTITY_LOCATION_WINDOW,
+  IDENTITY_LOCATION_COOKIE,
+  IDENTITY_LOCATION_MANAGED, // this is created if necessary and stored in a cookie with the associated name
+  IDENTITY_LOCATION_LOCAL_STORAGE,
+  IDENTITY_LOCATION_SESSION_STORAGE,
+  IDENTITY_LOCATION_QUERY_STRING,
+}
+
+// IdentityFormat is the encoding of the string identity value
+export enum IdentityFormat {
+  IDENTITY_FORMAT_UNDEFINED,
+  IDENTITY_FORMAT_STRING,
+  IDENTITY_FORMAT_JSON,
+  IDENTITY_FORMAT_JWT,
+  IDENTITY_FORMAT_QUERY,
+  IDENTITY_FORMAT_SEMICOLON,
+}
+
+// Identity represents all the metadata for an identifier on the page
 export interface Identity {
-  type: string
-  variable: string
+
+  // DEPRECATED: type is a string representing the location of the identity on the page
+  type?: string // deprecated
+
+  // location on the page from which to retrieve identity information
+  location?: IdentityLocation
```

Comment:
what is the default then if null? should this be required?

Reply:
yeah should be required - just not there yet. will keep type